### PR TITLE
Release/2.3.11

### PR DESCRIPTION
**Pull request by HanlinMiao**
Some PRs we can merge before cutting the release:

- [x] https://github.com/nautobot/nautobot/pull/6494
- [x] https://github.com/nautobot/nautobot/pull/6497

### Added

- [#6231](https://github.com/nautobot/nautobot/issues/6231) - Added `nautobot.apps.utils.get_related_field_for_models()` helper function.
- [#6231](https://github.com/nautobot/nautobot/issues/6231) - Added optional `lookup` parameter to `LinkedCountColumn`.

### Changed

- [#5321](https://github.com/nautobot/nautobot/issues/5321) - For bulk delete all objects view, only show the confirmation dialog without the table that shows the objects that would be deleted.
- [#6231](https://github.com/nautobot/nautobot/issues/6231) - Changed most related-object-count table columns (e.g. the "Locations" column in a Prefix table) to, if only a single related record is present (e.g. a single Location is associated with a given Prefix), display that related record directly instead of just displaying `1`.

### Fixed

- [#6414](https://github.com/nautobot/nautobot/issues/6414) - Fixed layout bug in browsable REST API.
- [#6442](https://github.com/nautobot/nautobot/issues/6442) - Fixes an issue where GitLab CI pipelines fail using all versions of official Docker images.
- [#6453](https://github.com/nautobot/nautobot/issues/6453) - Fixed issue where interfaces cannot be removed/deleted from an Interface for Modules.
- [#6455](https://github.com/nautobot/nautobot/issues/6455) - Fixes two tests which always passed due to errors in their implementation. Ensured they provide value by checking against the correct results.
- [#6472](https://github.com/nautobot/nautobot/issues/6472) - Fixed incorrect placement of buttons in create and edit views.
- [#6472](https://github.com/nautobot/nautobot/issues/6472) - Fixed the panel width in multiple create and edit views.

### Documentation

- [#6485](https://github.com/nautobot/nautobot/issues/6485) - Added additional clarification for enabling request profiling via user profile.

### Housekeeping

- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Added an integration test to create a Manufacturer, DeviceType, LocationType, Location, Role and Device to test the create forms and select2 api form fields are working correctly.
- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Fixed incorrect assertion in core navbar integration tests.
- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Added helper functions to SeleniumTestCase to perform some common UI actions.

### CODE REVIEW

File: nautobot/docs/release-notes/version-2.3.md [hunk 1]
```diff
@@ -124,6 +124,37 @@ Various button groups in the "object list" and "object detail" views have been c
...
+
+- [#5321](https://github.com/nautobot/nautobot/issues/5321) - For bulk delete all objects view, only show the confirmation dialog without the table that shows the objects that would be deleted.
+- [#6231](https://github.com/nautobot/nautobot/issues/6231) - Changed most related-object-count table columns (e.g. the "Locations" column in a Prefix table) to, if only a single related record is present (e.g. a single Location is associated with a given Prefix), display that related record directly instead of just displaying `1`.
+
+### Fixed
+
+- [#6414](https://github.com/nautobot/nautobot/issues/6414) - Fixed layout bug in browsable REST API.
+- [#6442](https://github.com/nautobot/nautobot/issues/6442) - Fixes an issue where GitLab CI pipelines fail using all versions of official Docker images.
+- [#6453](https://github.com/nautobot/nautobot/issues/6453) - Fixed issue where interfaces cannot be removed/deleted from an Interface for Modules.
+- [#6455](https://github.com/nautobot/nautobot/issues/6455) - Fixes two tests which always passed due to errors in their implementation. Ensured they provide value by checking against the correct results.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:144]: Maybe move this one to `Housekeeping`?

**Review by glennmatthews — APPROVED** [review]

File: nautobot/docs/release-notes/version-2.3.md [hunk 2]
```diff
@@ -124,6 +124,41 @@ Various button groups in the "object list" and "object detail" views have been c
...
+
+### Changed
+
+- [#5321](https://github.com/nautobot/nautobot/issues/5321) - For bulk delete all objects view, only show the confirmation dialog without the table that shows the objects that would be deleted.
+- [#6231](https://github.com/nautobot/nautobot/issues/6231) - Changed most related-object-count table columns (e.g. the "Locations" column in a Prefix table) to, if only a single related record is present (e.g. a single Location is associated with a given Prefix), display that related record directly instead of just displaying `1`.
+
+### Fixed
+
+- [#6414](https://github.com/nautobot/nautobot/issues/6414) - Fixed layout bug in browsable REST API.
+- [#6442](https://github.com/nautobot/nautobot/issues/6442) - Fixes an issue where GitLab CI pipelines fail using all versions of official Docker images.
```
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.3.md:142]:
```suggestion
- [#6442](https://github.com/nautobot/nautobot/issues/6442) - Fixed an issue where GitLab CI pipelines fail using all versions of official Docker images.
```

File: nautobot/docs/release-notes/version-2.3.md [hunk 3]
```diff
@@ -124,6 +124,41 @@ Various button groups in the "object list" and "object detail" views have been c
...
+### Documentation
+
+- [#6485](https://github.com/nautobot/nautobot/issues/6485) - Added additional clarification for enabling request profiling via user profile.
+
+### Housekeeping
+
+- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Added an integration test to create a Manufacturer, DeviceType, LocationType, Location, Role and Device to test the create forms and select2 api form fields are working correctly.
+- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Fixed incorrect assertion in core navbar integration tests.
+- [#6449](https://github.com/nautobot/nautobot/issues/6449) - Added helper functions to SeleniumTestCase to perform some common UI actions.
+- [#6455](https://github.com/nautobot/nautobot/issues/6455) - Fixes two tests which always passed due to errors in their implementation. Ensured they provide value by checking against the correct results.
```
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.3.md:159]:
```suggestion
- [#6455](https://github.com/nautobot/nautobot/issues/6455) - Fixed two tests which always passed due to errors in their implementation. Ensured they provide value by checking against the correct results.
```

**Review by timizuoebideri1 — APPROVED** [review]

File: nautobot/docs/release-notes/version-2.3.md [hunk 4]
```diff
@@ -124,6 +124,41 @@ Various button groups in the "object list" and "object detail" views have been c
 As Django 3.2 has reached end-of-life, Nautobot 2.3 requires Django 4.2, the next long-term-support (LTS) version of Django. There are a number of changes in Django itself as a result of this upgrade; Nautobot App maintainers are urged to review the Django release-notes ([4.0](https://docs.djangoproject.com/en/4.2/releases/4.0/), [4.1](https://docs.djangoproject.com/en/4.2/releases/4.1/), [4.2](https://docs.djangoproject.com/en/4.2/releases/4.2/)), especially the relevant "Backwards incompatible changes" sections, to proactively identify any impact to their Apps.
 
 <!-- towncrier release notes start -->
+## v2.3.11 (2024-11-11)
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:127]:
```suggestion
## v2.3.11 (2024-11-12)
```

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**